ManPages/tsManPagesLibGUI/tsWxPoint.man

Help on module tsWxPoint:

NAME
    tsWxPoint

FILE
    ./tsLibGUI/tsWxPkg/src/tsWxPoint.py

DESCRIPTION
    tsWxPoint.py - Class to represent a point or position of a graphical
    object with integer x (horizontal) and y (vertical) properties.

CLASSES
    __builtin__.object
        Point
    
    class Point(__builtin__.object)
     |  A data structure for representing a point or position with integer x
     |  and y properties. Most places in wxPython that expect a wx.Point can
     |  also accept a (x,y) tuple.
     |  
     |  Methods defined here:
     |  
     |  Get(self)
     |      Return the x and y properties as a tuple.
     |  
     |  Set(self, x, y)
     |      Set both the x and y properties.
     |  
     |  __add__(self, pt)
     |      Add pt properties to this and return the result.
     |  
     |  __eq__(self, other)
     |      Test for equality of wx.Point objects.
     |  
     |  __getitem__(self, index)
     |      Under Construction.
     |  
     |  __iadd__(self, pt)
     |      Add pt to this object.
     |  
     |  __init__(self, x=0, y=0)
     |      Constructor. Create a wx.Point object.
     |  
     |  __isub__(self, pt)
     |      Subtract pt from this object.
     |  
     |  __len__(self)
     |      Return length of object.
     |  
     |  __ne__(self, other)
     |      Test for inequality of wx.Point objects.
     |  
     |  __nonzero__(self)
     |      Return True if not zero.
     |  
     |  __reduce__(self)
     |      Under Construction.
     |  
     |  __repr__(self)
     |      Return text representation.
     |  
     |  __setitem__(self, index, val)
     |      Under Construction.
     |  
     |  __str__(self)
     |      Return text representation.
     |  
     |  __sub__(self, pt)
     |      Subtract pt properties from this and return the result
     |  
     |  tsDel_x(self)
     |  
     |  tsDel_y(self)
     |  
     |  tsGet_x(self)
     |  
     |  tsGet_y(self)
     |  
     |  tsSet_x(self, value)
     |  
     |  tsSet_y(self, value)
     |  
     |  ----------------------------------------------------------------------
     |  Static methods defined here:
     |  
     |  tsGetPointType(*args)
     |      Generate the specified class instance from the specified tuple.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors defined here:
     |  
     |  __dict__
     |      dictionary for instance variables (if defined)
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  x
     |  
     |  y

DATA
    DEBUG = True
    VERBOSE = True
    __authors__ = 'Richard S. Gordon'
    __copyright__ = 'Copyright (c) 2007-2013 Richard S. Gordon.\n\t\tAll r...
    __credits__ = '\n\n  Credits: \n\n\t  tsLibGUI Import & Application .....
    __date__ = '04/01/2013'
    __header__ = '\n\ntsWxPoint, v1.1.0 (build 04/01/2013)\n\n  Author... ...
    __license__ = 'GNU General Public License, Version 3, 29 June 2007'
    __line1__ = 'tsWxPoint, v1.1.0 (build 04/01/2013)'
    __line2__ = 'Author(s): Richard S. Gordon'
    __line3__ = 'Copyright (c) 2007-2013 Richard S. Gordon.\n\t\tAll right...
    __line4__ = 'GNU General Public License, Version 3, 29 June 2...U Gene...
    __title__ = 'tsWxPoint'
    __version__ = '1.1.0'
    mainTitleVersionDate = 'tsWxPoint, v1.1.0 (build 04/01/2013)'

VERSION
    1.1.0

DATE
    04/01/2013

CREDITS
    
    
      Credits: 
    
    	  tsLibGUI Import & Application Launch Features: 
    	  Copyright (c) 2007-2009 Frederick A. Kier.
    			All rights reserved.
    
    	  Python Curses Module API & Run Time Library Features:
    	  Copyright (c) 2001-2013 Python Software Foundation.
    			All rights reserved.
    	  PSF License Agreement for Python 2.7.3 & 3.3.0
    
    	  wxWidgets (formerly wxWindows) & wxPython API Features:
    	  Copyright (c) 1992-2008 Julian Smart, Robert Roebling,
    			Vadim Zeitlin and other members of the 
    			wxWidgets team.
    			All rights reserved.
    	  wxWindows Library License
    
    	  nCurses API & Run Time Library Features:
    	  Copyright (c) 1998-2011 Free Software Foundation, Inc.
    			All rights reserved.
    	  GNU General Public License, Version 3, 29 June 2007




Documents/NEWS.txt
#-----------------------------------------------------------
#"Time-stamp: <07/08/2015  4:48:41 AM rsg>"
#-----------------------------------------------------------

========================= NEWS.txt =========================

   +----+----+  TeamSTARS "tsWxGTUI_PyVx" Toolkit
   | ts | Wx |      with Python 2x & Python 3x based
   +----+----+         Command Line Interface (CLI)
   | G T U I |      and "Curses"-based "wxPython"-style, 
   +---------+         Graphical-Text User Interface (GUI)
 
   Get that cross-platform, pixel-mode "wxPython" feeling
   on character-mode 8-/16-color (xterm-family) & non-color
   (vt100-family) terminals and terminal emulators.

   You can find this and other plain-text files in the
   Toolkit subdirectory named:

       "./<Toolkit Recipient's Repository>/Documents".

       <Your Working Repository>
       (e.g. "tsWxGTUI_PyVx_Repository") 
         |
         |
         +-- ["Documents"]

=========================== News ===========================

The pre-alpha release of the TeamSTARS "tsWxGTUI_PyVx-0.0.0"
Toolkit is now available. It includes the following two
versions abd their associated variants (Python Distribution
Utilities ("Distutils") installable "Site-Packages" and non-
installable "Developer-Sandboxes" for enhancement and bug
fix development and experimentation without interference
with the already installed Site-Packages):

1. The TeamSTARS "tsWxGTUI_Py2x-0.0.0" Toolkit is released
   for users of the popular, high-level, cross-platform,
   second generation Python 2.x programming language which
   is widely available.

   A wiki posting by the Python Software Foundation announc-
   ed that Python 3.0 was released in 2008. The final 2.x
   version 2.7 release came out in mid-2010, with a state-
   ment of extended support for this end-of-life release.
   The 2.x branch will see no new major releases after that.

   The source code and documentation is available for down-
   load at:

      http://pypi.python.org/pypi.

   It supports development and operation of:

   a. Command Line User Interface (CLI) application
      programs implemented in Python 2.0.0-2.7.9.

   b. Graphical-style User Interface (GUI) application
      programs implemented in Python 2.6.4-2.7.9.

2. The TeamSTARS "tsWxGTUI_Py3x-0.0.0" Toolkit is released
   for users of the popular, high-level, cross-platform,
   third generation Python 3.x programming language which
   is growing in availability.

   A wiki posting by the Python Software Foundation announc-
   ed that Python 3.0 was released in 2008. 3.x is under
   active development and has already seen over five years
   of stable releases, including version 3.3 in 2012 and
   3.4 in 2014. This means that all recent standard library
   improvements, for example, are only available by default
   in Python 3.x.

   Guido van Rossum (the original creator of the Python
   language) decided to clean up Python 2.x properly,
   with less regard for backwards compatibility than is
   the case for new releases in the 2.x range. The most
   drastic improvement is the better Unicode support
   (with all text strings being Unicode by default) as
   well as saner bytes/Unicode separation.

   Besides, several aspects of the core language (such as
   print and exec being statements, integers using floor
   division) have been adjusted to be easier for newcomers
   to learn and to be more consistent with the rest of the
   language, and old cruft has been removed (for example,
   all classes are now new-style, "range()" returns a memory
   efficient iterable, not a list as in 2.x).

   The source code and documentation is available for down-
   load at:

      http://pypi.python.org/pypi.

   It supports development and operation of:

   a. Command Line User Interface (CLI) application
      programs implemented in Python 3.0.0-3.4.2.

   b. Graphical-style User Interface (GUI) application
      programs implemented in Python 3.1.5-3.4.2.

======================= End-Of-File ========================


SourceDistributions/Developer-Sandboxes/tsWxGTUI_PyVx/Python-3x/tsWxGTUI_Py3x/tsLibGUI/tsWxPkg/src/tsWxPoint.py
#! /usr/bin/env python
# "Time-stamp: <10/25/2013  6:41:47 AM rsg>"
'''
tsWxPoint.py - Class to represent a point or position of a graphical
object with integer x (horizontal) and y (vertical) properties.
'''
#################################################################
#
# File: tsWxPoint.py
#
# Purpose:
#
#    Class to represent a point or position of a graphical object
#    with integer x (horizontal) and y (vertical) properties.
#
# Usage (example):
#
#    # Import
#
#    from tsWxPoint import Point
#
# Requirements:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Capabilities:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Limitations:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Notes:
#
#    None
#
# Methods:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Classes:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Modifications:
#
#    None.
#
# ToDo:
#
#    None.
#
#################################################################

__title__     = 'tsWxPoint'
__version__   = '1.1.0'
__date__      = '04/01/2013'
__authors__   = 'Richard S. Gordon'
__copyright__ = 'Copyright (c) 2007-2013 ' + \
                '%s.\n\t\tAll rights reserved.' % __authors__
__license__   = 'GNU General Public License, ' + \
                'Version 3, 29 June 2007'
__credits__   = '\n\n  Credits: ' + \
                '\n\n\t  tsLibGUI Import & Application Launch Features: ' + \
                '\n\t  Copyright (c) 2007-2009 Frederick A. Kier.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\n\t  Python Curses Module API & ' + \
                'Run Time Library Features:' + \
                '\n\t  Copyright (c) 2001-2013 ' +\
                'Python Software Foundation.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  PSF License Agreement for Python 2.7.3 & 3.3.0' + \
                '\n\n\t  wxWidgets (formerly wxWindows) & ' + \
                'wxPython API Features:' + \
                '\n\t  Copyright (c) 1992-2008 Julian Smart, ' + \
                'Robert Roebling,' + \
                '\n\t\t\tVadim Zeitlin and other members of the ' + \
                '\n\t\t\twxWidgets team.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  wxWindows Library License' + \
                '\n\n\t  nCurses API & Run Time Library Features:' + \
                '\n\t  Copyright (c) 1998-2011 ' + \
                'Free Software Foundation, Inc.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  GNU General Public License, ' + \
                'Version 3, 29 June 2007'

__line1__ = '%s, v%s (build %s)' % (__title__, __version__, __date__)
__line2__ = 'Author(s): %s' % __authors__
__line3__ = '%s' % __copyright__

if len( __credits__) == 0:
    __line4__ = '%s' % __license__
else:
    __line4__ = '%s%s' % (__license__, __credits__)

__header__ = '\n\n%s\n\n  %s\n  %s\n  %s\n' % (__line1__,
                                               __line2__,
                                               __line3__,
                                               __line4__)

mainTitleVersionDate = __line1__

import tsWxGlobals as wx
import tsExceptions as tse

#---------------------------------------------------------------------------

##DEBUG = True
DEBUG = wx.Debug_GUI_Launch | \
        wx.Debug_GUI_Progress | \
        wx.Debug_GUI_Termination | \
        wx.Debug_GUI_Exceptions

##VERBOSE = True
VERBOSE = wx.Debug_GUI_Configuration

#---------------------------------------------------------------------------

class Point(object):
    '''
    A data structure for representing a point or position with integer x
    and y properties. Most places in wxPython that expect a wx.Point can
    also accept a (x,y) tuple.
    '''
    def __init__(self, x=0, y=0):
        '''
        Constructor. Create a wx.Point object.
        '''
##        theClass = 'Point'
##        # Create class membership flag self.thisown
##        self.tsRegisterClassNameAndMembershipFlag(theClass)
        self.ts_x = x
        self.ts_y = y

    #-----------------------------------------------------------------------

    def __add__(self, pt):
        '''
        Add pt properties to this and return the result.
        '''
        temp = Point.tsGetPointType(pt)
        return (Point(self.ts_x + temp.x, self.ts_y + temp.y))



    #-----------------------------------------------------------------------

##    def __del__(self):
##        '''
##        '''
##        del self.ts_x
##        del self.ts_y
##        del self

    #-----------------------------------------------------------------------

    def __eq__(self, other):
        '''
        Test for equality of wx.Point objects.
        '''
        temp = Point.tsGetPointType(other)
        return ((self.ts_x == temp.x) and \
                (self.ts_y == temp.y))
    #-----------------------------------------------------------------------

    def __getitem__(self, index):
        '''
        Under Construction.
        '''
        msg = 'NotImplementedError: %s' % '__getitem__ in tsWxPoint'
        # self.logger.error(msg)
        raise tse.ProgramException(tse.APPLICATION_TRAP, msg)

    #-----------------------------------------------------------------------

    def __iadd__(self, pt):
        '''
        Add pt to this object.
        '''
        return (Point(self.ts_x + pt.x, self.ts_y + pt.y))

    #-----------------------------------------------------------------------

    def __isub__(self, pt):
        '''
        Subtract pt from this object.
        '''
        return (Point(self.ts_x - pt.x, self.ts_y - pt.y))

    #-----------------------------------------------------------------------

    def __len__(self):
        '''
        Return length of object.
        '''
        return (len(self))

    #-----------------------------------------------------------------------

    def __ne__(self, other):
        '''
        Test for inequality of wx.Point objects.
        '''
        temp = Point.tsGetPointType(other)
        return ((self.ts_x != temp.x) or \
                (self.ts_y != temp.y))

    #-----------------------------------------------------------------------

    def __bool__(self):
        '''
        Return True if not zero.
        '''
        return (self.ts_x != 0 and \
                self.ts_y != 0)

    #-----------------------------------------------------------------------

    def __reduce__(self):
        '''
        Under Construction.
        '''
        msg = 'NotImplementedError: %s' % '__reduce__ in tsWxPoint'
        # self.logger.error(msg)
        raise tse.ProgramException(tse.APPLICATION_TRAP, msg)

    #-----------------------------------------------------------------------

    def __repr__(self):
        '''
        Return text representation.
        '''
        text = str((self.ts_x, self.ts_y))
        return (text)

    #-----------------------------------------------------------------------

    def __setitem__(self, index, val):
        '''
        Under Construction.
        '''
        msg = 'NotImplementedError: %s' % '__setitem__ in tsWxPoint'
        # self.logger.error(msg)
        raise tse.ProgramException(tse.APPLICATION_TRAP, msg)

    #-----------------------------------------------------------------------

    def __str__(self):
        '''
        Return text representation.
        '''
        return ('(%d, %d)' % (self.ts_x, self.ts_y))

    #-----------------------------------------------------------------------

    def __sub__(self, pt):
        '''
        Subtract pt properties from this and return the result
        '''
        temp = Point.tsGetPointType(pt)
        return (Point(self.ts_x - temp.x, self.ts_y - temp.y))

    #-----------------------------------------------------------------------

    def Get(self):
        '''
        Return the x and y properties as a tuple.
        '''
        return (self.ts_x, self.ts_y)

    #-----------------------------------------------------------------------

    def Set(self, x, y):
        '''
        Set both the x and y properties.
        '''
        self.ts_x = x
        self.ts_y = y

    #-----------------------------------------------------------------------

    @staticmethod
    def tsGetPointType(*args):
        '''
        Generate the specified class instance from the specified tuple.
        '''
        if ((len(args) == 2) and \
            (isinstance(args[0], int)) and \
            (isinstance(args[1], int))):

            return (Point(args[0], args[1]))

        elif ((len(args) == 1) and \
              (isinstance(args[0], Point))):

            return (args[0])

        elif ((len(args) == 1) and \
              (isinstance(args[0], tuple)) and \
              (len(args[0]) == 2)):

            theTuple = args[0]
            return (Point(theTuple[0], theTuple[1]))

        else:

##            fmt1 = 'Invalid data "%s" ' % str(theData)
##            fmt2 = 'with type "%s".' % str(type(theData))
##            msg = fmt1 + fmt2
##            print('***** ', msg)
##            raise tse.ProgramException(tse.APPLICATION_TRAP, msg)
            return (None)

    #-----------------------------------------------------------------------

    def tsGet_x(self): return self.ts_x
    def tsSet_x(self, value): self.ts_x = value
    def tsDel_x(self): del self.ts_x
    x = property(tsGet_x, tsSet_x, tsDel_x)

    #-----------------------------------------------------------------------

    def tsGet_y(self): return self.ts_y
    def tsSet_y(self, value): self.ts_y = value
    def tsDel_y(self): del self.ts_y
    y = property(tsGet_y, tsSet_y, tsDel_y)

#---------------------------------------------------------------------------

if __name__ == '__main__':

    print(__header__)

    wxPoint = Point

    thePoint = wxPoint(x=100, y=200)
    print('thePoint: %s; x: %d; y: %d' % (thePoint, thePoint.x, thePoint.y))
    thePoint.Set(thePoint.x * 3, thePoint.y * 4)
    print('thePoint: %s; x: %d; y: %d' % (thePoint, thePoint.x, thePoint.y))

    pt = wxPoint(x=10, y=20)
    print('pt: %s; x: %d; y: %d' % (pt, pt.x, pt.y))

    theSum = thePoint.__add__(pt)
    print('__add__: %s; x: %d; y: %d' % (theSum, theSum.x, theSum.y))

    theEq = thePoint.__eq__(thePoint)
    print('__eq__: %s; thePoint: %s; thePoint: %s' % (
        theEq, str(thePoint), str(thePoint)))

    theEq = thePoint.__eq__(pt)
    print('__eq__: %s; thePoint: %s; pt: %s' % (
        theEq, str(thePoint), str(pt)))

    theSum = thePoint.__iadd__(pt)
    print('__iadd__: %s; x: %d; y: %d' % (theSum, theSum.x, theSum.y))

    theNe = thePoint.__ne__(thePoint)
    print('__ne__: %s; thePoint: %s; thePoint: %s' % (
        theNe, str(thePoint), str(thePoint)))

    theNe = thePoint.__ne__(pt)
    print('__ne__: %s; thePoint: %s; pt: %s' % (
        theNe, str(thePoint), str(pt)))

    theSum = thePoint.__sub__(pt)
    print('__sub__: %s; x: %d; y: %d' % (theSum, theSum.x, theSum.y))

    theSum = thePoint.__isub__(pt)
    print('__isub__: %s; x: %d; y: %d' % (theSum, theSum.x, theSum.y))

    print('Get: %s; x: %d; y: %d' % (thePoint.Get(), thePoint.x, thePoint.y))

    thePoint.Set(x=500, y=600)
    print('Set: %s; x: %d; y: %d' % (thePoint.Get(), thePoint.x, thePoint.y))

    theInput = [1, 2]
    theInputType = str(type(theInput))
    theOutput = wxPoint.tsGetPointType(1, 2)
    print('\n\ntsGetPointType (%s): \n\ttheOutput=%s; theInput:=%s; %s' % (
        type(theOutput), theOutput, theInput, theInputType))

    theInput = wxPoint(2, 3)
    theInputType = str(type(theInput))
    theOutput = wxPoint.tsGetPointType(theInput)
    print('\n\ntsGetPointType (%s): \n\ttheOutput=%s; theInput:=%s; %s' % (
        type(theOutput), theOutput, theInput, theInputType))

    theInput = (3, 4)
    theInputType = str(type(theInput))
    theOutput = wxPoint.tsGetPointType(theInput)
    print('\n\ntsGetPointType (%s): \n\ttheOutput=%s; theInput:=%s %s' % (
        type(theOutput), theOutput, theInput, theInputType))

    theInput = [4, 5]
    theInputType = str(type(theInput))
    theOutput = wxPoint.tsGetPointType(theInput)
    print('\n\ntsGetPointType (%s): \n\ttheOutput=%s; theInput:=%s %s' % (
        type(theOutput), theOutput, theInput, theInputType))

    theInput = (5, 6, 7)
    theInputType = str(type(theInput))
    theOutput = wxPoint.tsGetPointType(theInput)
    print('\n\ntsGetPointType (%s): \n\ttheOutput=%s; theInput:=%s %s' % (
        type(theOutput), theOutput, theInput, theInputType))

    theInput = (6)
    theInputType = str(type(theInput))
    theOutput = wxPoint.tsGetPointType(theInput)
    print('\n\ntsGetPointType (%s): \n\ttheOutput=%s; theInput:=%s %s' % (
        type(theOutput), theOutput, theInput, theInputType))

    theInput = (7.0)
    theInputType = str(type(theInput))
    theOutput = wxPoint.tsGetPointType(theInput)
    print('\n\ntsGetPointType (%s): \n\ttheOutput=%s; theInput:=%s %s' % (
        type(theOutput), theOutput, theInput, theInputType))

    theInput = 8
    theInputType = str(type(theInput))
    theOutput = wxPoint.tsGetPointType(theInput)
    print('\n\ntsGetPointType (%s): \n\ttheOutput=%s; theInput:=%s %s' % (
        type(theOutput), theOutput, theInput, theInputType))
